src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -1700,6 +1700,7 @@ private void emitNoResolvedType(NoType type, String defaultEmit) {
       if (importRenameMap.containsKey(displayName)) {
         displayName = importRenameMap.get(displayName);
       }
+      if (maybeEmitGlobalAlias(type)) return;
```

Comment:
Move this up to above `String displayName = type.getDisplayName();` - let's keep all the early return guards at the top.

Reply:
moved up.